### PR TITLE
PLATUI-468: Externalise Javascript content into Scala messages file

### PR DESCRIPTION
**Pull request by matthewmascord**
Adds functionality to import content from Scala Play! messages file into Javascript bundle using dot-properties.

N.B. this is not a full implementation of java.text.MessageFormat so it does not support placeholders as per https://docs.oracle.com/javase/7/docs/api/java/text/MessageFormat.html

### CODE REVIEW

File: app/uk/gov/hmrc/trackingconsentfrontend/assets/src/ui/banner.ts [hunk 1]
```diff
@@ -0,0 +1,28 @@
+const banner = messages => `<div class="cbanner-govuk-width-container cbanner-cookie-banner__question">
+    <div class="cbanner-govuk-grid-row">
+        <div class="cbanner-govuk-grid-column-full">
+            <span class="cbanner-govuk-heading-m">${messages['banner.title']}</span>
```
Comment (timsb) [inline app/uk/gov/hmrc/trackingconsentfrontend/assets/src/ui/banner.ts:4]: Should this be a heading? Class and message key suggests so.

File: app/uk/gov/hmrc/trackingconsentfrontend/assets/src/ui/banner.ts [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+                </a>
+                ${messages['banner.paragraph1.afterLink']}
+            <div>
+                <div class="cbanner-cookie-banner__button-column cbanner-govuk-grid-column-full cbanner-govuk-grid-column-one-half-from-desktop">
+                    <button class="cbanner-govuk-button cbanner-cookie-banner__accept-all-button" type="submit">
+                        ${messages['banner.acceptAllButton']}
+                    </button>
+                </div>
+                <div class="cbanner-cookie-banner__button-column cbanner-govuk-grid-column-full cbanner-govuk-grid-column-one-half-from-desktop">
+                    <a href="/tracking-consent/cookie-settings" class="cbanner-govuk-button cbanner-cookie-banner__set-cookie-preferences-button" draggable="false">
```
Comment (timsb) [inline app/uk/gov/hmrc/trackingconsentfrontend/assets/src/ui/banner.ts:18]: Is this something that looks like a button? If so, should have `role=button` and be able to be interacted with using the space bar.

File: app/uk/gov/hmrc/trackingconsentfrontend/assets/src/ui/confirmation.ts [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+<p class="cbanner-govuk-body cbanner-cookie-banner__confirmation-message-text">
+    ${messages['banner.saveConfirmation.paragraph1.beforeLink']}
+    <a class="cbanner-govuk-link" href="${messages['banner.saveConfirmation.paragraph1.link.url']}">
+        ${messages['banner.saveConfirmation.paragraph1.link.text']}
+    </a>
+    ${messages['banner.saveConfirmation.paragraph1.afterLink']}
+</p>
+<p class="cbanner-govuk-body cbanner-cookie-banner__hide-button-container">
+    <button class="cbanner-govuk-link cbanner-cookie-banner__hide-button" role="button">
+            ${messages['banner.saveConfirmation.hideButton']}
```
Comment (timsb) [inline app/uk/gov/hmrc/trackingconsentfrontend/assets/src/ui/confirmation.ts:11]: A screen reader user wouldn't necessarily understand the context of the word 'hide'. Giving more context or e.g. an aria label would help.

File: app/uk/gov/hmrc/trackingconsentfrontend/assets/src/ui/confirmation.ts [hunk 4]
```diff
@@ -0,0 +1,15 @@
+const confirmation = messages => `
+<p class="cbanner-govuk-body cbanner-cookie-banner__confirmation-message-text">
+    ${messages['banner.saveConfirmation.paragraph1.beforeLink']}
+    <a class="cbanner-govuk-link" href="${messages['banner.saveConfirmation.paragraph1.link.url']}">
+        ${messages['banner.saveConfirmation.paragraph1.link.text']}
+    </a>
+    ${messages['banner.saveConfirmation.paragraph1.afterLink']}
+</p>
+<p class="cbanner-govuk-body cbanner-cookie-banner__hide-button-container">
+    <button class="cbanner-govuk-link cbanner-cookie-banner__hide-button" role="button">
```
Comment (timsb) [inline app/uk/gov/hmrc/trackingconsentfrontend/assets/src/ui/confirmation.ts:10]: `role=button` isn't needed here.

File: app/uk/gov/hmrc/trackingconsentfrontend/assets/src/loaders/message-format-loader/index.js [hunk 5]
```diff
@@ -0,0 +1,21 @@
+const {parse} = require('dot-properties')
+
+const stripDoubleQuotesFromString = content => content.replace(/''/, "'")
```
Comment (timsb) [inline app/uk/gov/hmrc/trackingconsentfrontend/assets/src/loaders/message-format-loader/index.js:3]: Typographic quotes should be used in services which would mean this isn't necessary.

Comment (matthewmascord) [reply app/uk/gov/hmrc/trackingconsentfrontend/assets/src/loaders/message-format-loader/index.js:3]: This isn't related to that, it's to properly translate escaped single quotes in Scala message format files.

Comment (timsb) [reply app/uk/gov/hmrc/trackingconsentfrontend/assets/src/loaders/message-format-loader/index.js:3]: Yeah but I don't think there should be any normal straighty quotes in your messages file, only curly/typographic quotes.

Comment (matthewmascord) [reply app/uk/gov/hmrc/trackingconsentfrontend/assets/src/loaders/message-format-loader/index.js:3]: Fixed.

**Pull request merged**